Bikes/Bikes.Infrastructure.InMemory/Repositories/InMemoryBikeModelRepository.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Bikes.Domain.Models;
+using Bikes.Domain.Repositories;
+using Bikes.Infrastructure.InMemory.Seeders;
+
+namespace Bikes.Infrastructure.InMemory.Repositories;
+
+/// <summary>
+/// Implementing a repository for the BikeModel class 
+/// </summary>
+public class InMemoryBikeModelRepository : IRepository<BikeModel, int>
+{
+    private readonly List<BikeModel> _items;
+
+    private int _currentId;
+
+    /// <summary>
+    /// A constructor that uses data from InMemorySeeder
+    /// </summary>
+    public InMemoryBikeModelRepository()
+    {
+        _items = InMemorySeeder.GetBikeModels();
+        _currentId = _items.Count > 0 ? _items.Count : 0;
+    }
+
+    /// <summary>
+    /// Creates a new object
+    /// </summary>
+    /// <param name="entity">Object</param>
+    /// <returns>ID of the created object</returns>
+    public int Create(BikeModel entity)
+    {
+        entity.Id = ++_currentId;
+        _items.Add(entity);
+        return entity.Id;
+    }
+
+    /// <summary>
+    /// Returns all existing objects
+    /// </summary>
+    /// <returns>List of existing objects</returns>
+    public List<BikeModel> ReadAll()
+    {
+        return _items;
```

Comment:
Не очень безопасно отдавать саму коллекцию. А если кто то в обход методов репозитория решит этот список обрабатывать. Лучше отдавайте копию. Аналогично поправить в других репозиториях.
```suggestion
        return [.. _items];
```